File: Tutoriels/Cpp/Projets/src/manager/GProcessConfig.cpp
//===============================================
#include "GProcessConfig.h"
#include "GConfig.h"
//===============================================
GProcessConfig* GProcessConfig::m_instance = 0;
//===============================================
GProcessConfig::GProcessConfig() {

}
//===============================================
GProcessConfig::~GProcessConfig() {

}
//===============================================
GProcessConfig* GProcessConfig::Instance() {
    if(m_instance == 0) {
        m_instance = new GProcessConfig;
    }
    return m_instance;
}
//===============================================
void GProcessConfig::run(int argc, char **argv) {
    GConfig::Instance()->setData("Nom", "KESSE");
    GConfig::Instance()->setData("Prenom", "Gerard");
    GConfig::Instance()->setData("Email", "gerard.kesse@readydev.com");
    GConfig::Instance()->setData("Diplome", "Ingenieur");
    GConfig::Instance()->setData("Formation", "Informatique Industrielle");
    GConfig::Instance()->setData("Ecole", "Polytech'Montpellier");
    GConfig::Instance()->show();
}
//===============================================

File: Tutoriels/CSharp/Projets/src/manager/GProcessConfig.cs
//===============================================
using System;
//===============================================
public class GProcessConfig : GProcess {
    //===============================================
    private static GProcessConfig m_instance = null;
    //===============================================
    private GProcessConfig() {

    }
    //===============================================
    new public static GProcessConfig Instance() {
        if(m_instance == null) {
            m_instance = new GProcessConfig();
        }
        return m_instance;
    }
    //===============================================
    public override void Run(string[] args) {
        GConfig.Instance().SetData("Nom", "KESSE");
        GConfig.Instance().SetData("Prenom", "Gerard");
        GConfig.Instance().SetData("Email", "gerard.kesse@readydev.com");
        GConfig.Instance().SetData("Diplome", "Ingenieur");
        GConfig.Instance().SetData("Formation", "Informatique Industrielle");
        GConfig.Instance().SetData("Ecole", "Polytech'Montpellier");
        GConfig.Instance().ShowData();
    }
    //===============================================
}
//===============================================
